vizro-core/docs/pages/user-guides/data.md

```diff
@@ -268,3 +265,81 @@ data_manager["slow_expire_data"].timeout = 60 * 60
 data_manager["no_expire_data"] = load_iris_data
 data_manager["no_expire_data"].timeout = 0
 ```
+
+### Parametrize data loading
+
+You can supply arguments to your dynamic data loading function that can be modified from the dashboard. For example, if you are handling big data then you might like to load on demand only one chunk of the data.
+
+To add a parameter to control a dynamic data source, do the following:
+
+1. add the appropriate argument to your dynamic data function and specify a default value for the argument.
+2. give an `id` to all components that have the data source you wish to alter through a parameter.
+3. [add a parameter](parameters.md) with `targets` of the form `<target_component_id>.data_frame.<dynamic_data_argument>` and a suitable [selector](selectors.md).
+
+For example, let us extend the [dynamic data example](#dynamic-data) above to show how the `load_iris_data` can take an argument `number_of_points` controlled from the dashboard with a [`Slider`][vizro.models.Slider].
+
+!!! example "Parametrized dynamic data"
+    === "app.py"
+        ```py hl_lines="8 10 20-23"
+        from vizro import Vizro
+        import pandas as pd
+        import vizro.plotly.express as px
+        import vizro.models as vm
+
+        from vizro.managers import data_manager
+
+        def load_iris_data(number_of_points=10): # (1)!
+            iris = pd.read_csv("iris.csv") # (2)!
+            return iris.sample(number_of_points) # (3)!
+
+        data_manager["iris"] = load_iris_data # (4)!
+
+        page = vm.Page(
+            title="Update the chart on page refresh",
+            components=[
+                vm.Graph(id="graph", figure=px.box("iris", x="species", y="petal_width", color="species")) # (5)!
```

Comment:
> I would suggest a different plot, one where you can more easily see the number of dots.

I totally agree with you. However, this change would require changes in several more examples/images/gifs in the same page. I would rather keep it as it is and then (in a separate PR) make advanced examples for the entire `data.md` page, moreover because there is already a ticket for that. https://github.com/McK-Internal/vizro-internal/issues/715

---

> I would also make the slider prettier, with a smaller step size

How about this selector? 
`vm.Slider(min=10, max=100, step=10, value=10)` 
<img width="310" alt="image" src="https://github.com/mckinsey/vizro/assets/108530920/1acb9111-aabc-48b0-9073-3689fd2707a9">